helix-term/src/ui/editor.rs

```diff
@@ -695,6 +750,342 @@ impl EditorView {
+        theme: &Theme,
+    ) {
+        let Some(context) = context else {
+            return;
+        };
+
+        let text = doc.text().slice(..);
+        let viewport = view.inner_area(doc);
+
+        // backup (status line) shall always exist
+        let status_line_style = theme
+            .try_get("ui.statusline.context")
+            .expect("`ui.statusline.context` exists");
+
+        // define sticky context styles
+        let context_style = theme
+            .try_get("ui.sticky.context")
+            .unwrap_or(status_line_style);
+        let indicator_style = theme
+            .try_get("ui.sticky.indicator")
+            .unwrap_or(status_line_style);
+
+        let mut context_area = viewport;
+        context_area.height = 1;
+
+        for node in context {
+            surface.clear_with(context_area, context_style);
+
+            let mut new_offset = view.offset;
+            let mut line_context_area = context_area;
+
+            if let Some(indicator) = node.indicator.as_deref() {
+                // set the indicator
+                surface.set_stringn(
+                    line_context_area.x,
+                    line_context_area.y,
+                    indicator,
+                    indicator.len(),
+                    indicator_style,
+                );
+                continue;
+            }
+
+            // get the len of bytes of the text that will be written (the "definition" line)
+            let line = text.line(node.line);
+            let tab_count = line.chars().take_while(|c| *c == '\t').count();
+
+            let already_written =
+                (line.len_bytes() + tab_count.saturating_mul(doc.tab_width() - 1)) as u16;
+
+            let dots = "...";
+
+            let virtual_text_annotations = TextAnnotations::default();
+
+            // if the definition of the function contains multiple lines
+            if node.has_context_end {
+                let last_line = text.byte_to_line(node.byte_range.end);
+                let overdraw_offset = text
+                    .line(last_line)
+                    .chars()
+                    .take_while(|c| c.is_whitespace())
+                    .count()
+                    + 1;
+
+                // handle tabs
+                let overdraw_offset =
+                    overdraw_offset + tab_count.saturating_mul(doc.tab_width() - 1);
+
+                // calculation of the correct space on where the end of the signature
+                // should be drawn at
+                let mut additional_area = line_context_area;
+                additional_area.x +=
+                    (already_written + dots.len() as u16).saturating_sub(overdraw_offset as u16);
+
+                // render the end of the function definition
+                let mut renderer = TextRenderer::new(
+                    surface,
+                    doc,
+                    theme,
+                    view.offset.horizontal_offset,
```

Comment:
I think you should add the start column of `context.end`  to the horizontal offset here instead of subtracting the overdraw offset above